app/src/components/fields/TelemetrySelectField.tsx

```diff
@@ -0,0 +1,55 @@
+import { FormControl, FormControlProps, FormHelperText, InputLabel, MenuItem, Select } from '@mui/material';
+import { useFormikContext } from 'formik';
+import useDataLoader from 'hooks/useDataLoader';
+import get from 'lodash-es/get';
+import React from 'react';
+
+interface ITelemetrySelectField {
+  name: string;
+  label: string;
+  id: string;
+  fetchData: () => Promise<any>;
+  controlProps?: FormControlProps;
+}
+
+interface ISelectOption {
+  value: string | number;
+  label: string;
+}
+
+const TelemetrySelectField: React.FC<ITelemetrySelectField> = (props) => {
+  const bctwLookupLoader = useDataLoader(() => props.fetchData());
+  const { values, touched, errors, handleChange, handleBlur } = useFormikContext<ISelectOption>();
+
+  const err = get(touched, props.name) && get(errors, props.name);
+
+  if (!bctwLookupLoader.data) {
+    bctwLookupLoader.load();
+  }
+
+  return (
+    <FormControl variant="outlined" fullWidth {...props.controlProps} error={!!err}>
+      <InputLabel id={`${props.name}-label`}>{props.label}</InputLabel>
+      <Select
+        name={props.name}
+        labelId="telemetry_select"
+        label={props.label}
+        value={get(values, props.name) ?? ''}
+        onChange={handleChange}
+        onBlur={handleBlur}
+        displayEmpty
+        inputProps={{ 'aria-label': 'Permit Type' }}>
+        {bctwLookupLoader.data?.map((a: any) => {
+          return (
+            <MenuItem key={a} value={a}>
+              {a}
```

Comment:
This basically asserts that each element in `bctwLookupLoader.data` is going to be a `string`. Could you provide types for `props.fetchData` accordingly?